pkg/apis/application/v1alpha1/types.go

```diff
@@ -111,6 +113,10 @@ type ApplicationSourceKsonnet struct {
 	Environment string `json:"environment,omitempty" protobuf:"bytes,1,opt,name=environment"`
 }
 
+type ApplicationSourceDirectory struct {
+	Recurse bool `json:"recurse" protobuf:"bytes,1,opt,name=recurse"`
```

Comment:
We should add the omitempty tag so payload only will include recurse when set to true.
```
	Recurse bool `json:"recurse,omitempty" protobuf:"bytes,1,opt,name=recurse"`
```